### PR TITLE
Add Makefile for Netlify to use to build docs

### PR DESCRIPTION
**Pull request by jputrino**
### Proposed changes

This pull request sets up the repo to use Netlify to build and host deploy previews for the IC documentation.

Next steps:
- Netlify support will activate automatic builds for this repo once this addition is merged to master.
- Netlify will automatically build docs and create deploy previews for all pull requests into master and release branches.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x ] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [ N/A] I have added tests that prove my fix is effective or that my feature works
- [ N/A] I have checked that all unit tests pass after adding my changes
- [N/A ] I have updated necessary documentation
- [ x] I have rebased my branch onto master
- [ x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: docs-web/Makefile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#docker-all: docker setup build
+
+clean:
+	rm -rf ${SITE_ROOT}
+
+setup:
+	mkdir ${SITE_ROOT}
+	@git clone -q --single-branch --branch master https://$(GITHUB_PAT)@github.com/nginxinc/docs-nginx-com.git "${SITE_ROOT}"/docs-nginx-com
+	pip install -r ${SITE_ROOT}/docs-nginx-com/requirements.txt
+	rsync -avOz "${NETLIFY_ROOT}"/docs-web/ ${SITE_ROOT}/docs-nginx-com/source/nginx-ingress-controller/ --exclude ${NETLIFY_ROOT}/docs-web/.netlify --exclude ${NETLIFY_ROOT}/docs-web/Makefile
```
Comment (pleshakov) [inline docs-web/Makefile:20]: `--exclude ${NETLIFY_ROOT}/docs-web/.netlify` - this file doesn't exist in our repo, right?

File: docs-web/Makefile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+NETLIFY_ROOT	=	/opt/build/repo
```
Comment (pleshakov) [inline docs-web/Makefile:1]: could we possibly use the same assignment style as in the https://github.com/nginxinc/kubernetes-ingress/blob/master/Makefile ?
For example,
```
NETLIFY_ROOT = /opt/build/repo
```

File: docs-web/Makefile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+SITE_ROOT			= ${NETLIFY_ROOT}/site
+BUILDDIR			= ${NETLIFY_ROOT}/docs-web/build
+#DOCKER_IMAGE 	=
```
Comment (pleshakov) [inline docs-web/Makefile:4]: the Makefile includes a few commented out parts. Are those necessary to keep or could be removed?

File: docs-web/Makefile (hunk 1 above)
Comment (jputrino) [reply docs-web/Makefile]: Correct! Removing.

File: docs-web/Makefile [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+.PHONY: all build clean setup
+
+all: clean setup build
+
+clean:
+	rm -rf ${SITE_ROOT}
+
+setup:
+	mkdir ${SITE_ROOT}
```
Comment (lucacome) [inline docs-web/Makefile:14]:
```suggestion
	mkdir -p ${SITE_ROOT}
```

Comment (jputrino) [reply docs-web/Makefile:14]: @lucacome Thanks! The Netlify build environment always clones the repo into the path `/opt/build/repo`. If we're just adding another directory in that existing location, do we need to use the `-p`?

Comment (lucacome) [reply docs-web/Makefile:14]: It's just safer, if the directory already exists or if part of the path doesn't exist, it won't throw an error. But not strictly necessary, that's why I approved anyway 🙂

Comment (martialonline) [reply docs-web/Makefile:14]: I agree with @lucacome here, `-p` is considered good practice and certainly won't do any harm but it's not necessarily needed.

Comment (jputrino) [reply docs-web/Makefile:14]: Thanks guys!

**Pull request closed without merging**